### DISCUSSION TITLE
Z80 VDC and Z80 KBDMSE build and test

### DISCUSSION THREAD

**Discussion by lynchaj** [lynchaj/nhyodyne#3] [General]
Hi,
Build and test for Z80 VDC and Z80 KBDMSE is coming along well.  Both seem to be working although Z80 VDC will definitely need some mods so there will have to be a Z80 VDC V2.  Z80 KBDMSE works unmodified but definitely expects a VT82C42 and a scrap 80C42 is not sufficient alternative.

If you are seeing this message please give a short reply so I can get a sense as to who can see the discussion forums

Thanks!  Andrew Lynch

**Comment by wwarthen** [comment]
Hi Andrew,

Yes, I saw your message.  I am signed up to "Watch" all activity on this repository which may be why I was notified when your message was posted.

Thanks,

Wayne

**Comment by lynchaj** [comment]
Hi
I've built and tested the Z80 VDC and Z80 KBDMSE and they seem to work fine.  The Z80 VDC works with the GBS-8200 fairly well but requires the pseudo-VGA circuit for analog color signals and a small mod for CSYNC so there will definitely be a PCB respin for Z80 VDC.

Same thing for Z80 KBDMSE; the keyboard circuit works fine but it requires a VT82C42N controller and there doesn't seem to be any substitute.  I thought maybe a regular AT 8042 might work but it doesn't.  Also learned a lot about the PS/2 mouse circuit.  It seems to work fine but due to some odd characteristics of the 8242 it will need two separate interrupt lines to distinguish between keyboard and mouse input.  

The current board has two interrupts but they're OR'd together and there is no way to determine which device issued the interrupt (?) so we're going to need two separate interrupts and use the IM2 circuit on the Z80 processor board to separate them.  Also needs some diagnostic LEDs to help writing software because the 8242 is very confusing.

Thanks, Andrew Lynch
